staging/src/k8s.io/apimachinery/pkg/util/net/interface_test.go

```diff
@@ -575,24 +582,29 @@ func TestGetIPFromHostInterfaces(t *testing.T) {
 	testCases := []struct {
 		tcase      string
 		nw         networkInterfacer
+		order      []AddressFamily
 		expected   net.IP
 		errStrFrag string
 	}{
-		{"fail get I/Fs", failGettingNetworkInterface{}, nil, "failed getting all interfaces"},
-		{"no interfaces", noNetworkInterface{}, nil, "no interfaces"},
-		{"I/F not up", downNetworkInterface{}, nil, "no acceptable"},
-		{"loopback only", loopbackNetworkInterface{}, nil, "no acceptable"},
-		{"P2P I/F only", p2pNetworkInterface{}, nil, "no acceptable"},
-		{"fail get addrs", networkInterfaceFailGetAddrs{}, nil, "unable to get Addrs"},
-		{"no addresses", networkInterfaceWithNoAddrs{}, nil, "no acceptable"},
-		{"invalid addr", networkInterfaceWithInvalidAddr{}, nil, "invalid CIDR"},
-		{"no matches", networkInterfaceWithOnlyLinkLocals{}, nil, "no acceptable"},
-		{"ipv4", validNetworkInterface{}, net.ParseIP("10.254.71.145"), ""},
-		{"ipv6", ipv6NetworkInterface{}, net.ParseIP("2001::200"), ""},
+		{"fail get I/Fs", failGettingNetworkInterface{}, preferV4, nil, "failed getting all interfaces"},
```

Comment:
should we add a test case for `0.0.0.0` and `::` addresses? 
it will cover liggitt 's question for the future

Reply:
IC, I was asking because looking at the current test cases I saw one test case with link locals
`networkInterfaceWithOnlyLinkLocals` despite it's discarded in the `ResolveBindAddress` too.